src/rules/data-quality/schedule-contains-recurrence-data-rule.js

```diff
@@ -0,0 +1,88 @@
+const { RRule } = require('rrule');
+const Rule = require('../rule');
+const generateRRuleOptions = require('../../helpers/rrule-options');
+const getScheduleProperties = require('../../helpers/schedule-properties');
+const ValidationErrorType = require('../../errors/validation-error-type');
+const ValidationErrorCategory = require('../../errors/validation-error-category');
+const ValidationErrorSeverity = require('../../errors/validation-error-severity');
+
+module.exports = class ValidRecurrenceRule extends Rule {
+  constructor(options) {
+    super(options);
+    this.targetModels = ['Schedule'];
+    this.meta = {
+      name: 'ValidRecurrenceRule',
+      description:
+        'Validates that the Schedule contains the correct information to generate a valid iCal recurrence rule.',
+      tests: {
+        default: {
+          message:
+            'Schedule must contains the correct information to generate a valid iCal recurrence rule.',
+          sampleValues: {
+            startTime: '08:30',
+            endTime: '09:30',
+            startDate: '2021-03-19',
+            repeatFrequency: 'P1W',
+            count: 10,
+            scheduleTimezone: 'Europe/London',
+          },
+          category: ValidationErrorCategory.CONFORMANCE,
+          severity: ValidationErrorSeverity.FAILURE,
+          type: ValidationErrorType.MISSING_REQUIRED_FIELD,
+        },
+        dtStart: {
+          message:
+            'The recurrence rule must contain a startDate, startTime, and scheduledTimezone to generate the schedule.',
+          sampleValues: {
+            startTime: '08:30',
+            startDate: '2021-03-19',
+            scheduleTimezone: 'Europe/London',
+          },
+          category: ValidationErrorCategory.CONFORMANCE,
+          severity: ValidationErrorSeverity.FAILURE,
+          type: ValidationErrorType.MISSING_REQUIRED_FIELD,
+        },
+      },
+    };
+  }
+
+  validateModel(node) {
+    const errors = [];
+
+    const properties = getScheduleProperties(node);
+    const rruleOptions = generateRRuleOptions(properties);
+
+    if (typeof properties.startDate === 'undefined'
+        || typeof properties.startTime === 'undefined'
+        || typeof properties.scheduleTimezone === 'undefined') {
+      errors.push(
+        this.createError('dtStart', {
+          value: undefined,
+          path: node,
+        }),
+      );
+    }
+
+    try {
+      const rule = new RRule(rruleOptions);
+      const firstEvent = rule.all()[0];
+      if (firstEvent.getTime() !== rruleOptions.dtstart.getTime()) {
+        errors.push(
+          this.createError('default', {
+            error: 'The first event in the generated by RRule does not match the Schedule data.',
```

Comment:
`error` is passed here but does not appear in `message`, so will not be displayed. Suggest this is actually a specific key of `test` above, that includes a `message` which specifically includes this string

It should also explain exactly what is being checked here

```suggestion
            error: 'The first event that is generated by the `Schedule` does not match the `startDate` and `startTime`.',
```